Project 2/k-NNAlgorithms/src/Algorithms.java

```diff
@@ -0,0 +1,171 @@
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class Algorithms {
+
+
+    //base KNN function
+    public static ArrayList<String> KNN(ArrayList<ArrayList<String>>trainingData, ArrayList<ArrayList<String>>testingData, int k, boolean regression, boolean euclidean){
+        //variables to hold sizes of things
+        int lengthOfTrainingSet = trainingData.size();
```

Comment:
Doesn't really matter in our case, but probably better practice to use a const/readonly instead of int.